joomla/joomla-cms
# [Accessibility] Category list

#### Description

This PR makes category list menu item type acessible according to WCAG 2.0.
1. List of articles table:
   - Now the table header always exists in the HTML, through CSS it show or hides according to "Show Headings" Show/Hide selection.
     
     See http://www.w3.org/TR/WCAG20-TECHS/H51.html
   - A `caption` tag (css hidden by default) is added to the Table.
     
     See http://www.w3.org/TR/WCAG20-TECHS/H39.html
   - Column scopes (`scope="col"`) are added to `th` tags.
     
     See http://www.w3.org/TR/WCAG20-TECHS/H63.html
2. Form with filters:
   - The form tag is now only in for the filter field (not form + table inside form tag like before).
   - A `legend` tag (css hidden by default) is added to the fieldset.
     
     See http://www.w3.org/TR/WCAG20-TECHS/H71.html
   - A submit button is added (css hidden by default) to the filter form.
     
     See http://www.w3.org/TR/WCAG20-TECHS/H32.html
#### How to test
1. Use a fresh joomla install with sample items.
2. Create a "Category list" menu item type and play with "List layout" separator options and check if everything is showing as it should in the frontend.
3. Use an external automatic acessibility checker to check if everything is ok. Some tools: http://www.w3.org/WAI/ER/tools/
#### More info

Improvements are welcome
